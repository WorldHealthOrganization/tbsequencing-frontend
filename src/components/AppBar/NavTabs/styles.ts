import { fontSizes } from '../../typography/fontSizes';
import appColors from '../../../styles/colors';
import { appZIndex } from '../../../styles/appZIndex';

export const sxTabStyles = {
  fontSize: fontSizes.h2,
  paddingTop: `${32}px`,
  paddingBottom: `${36}px`,
  color: appColors.primary.navy,
  '&.Mui-selected': {
    color: appColors.primary.navy,
    fontWeight: 'bold',
  },
  textTransform: 'capitalize',
};

export const sxTabsStyles = {
  '& .MuiTabs-indicator': {
    display: 'flex',
    justifyContent: 'center',
    backgroundColor: appColors.secondary.whoBlue,
    height: 4,
    zIndex: appZIndex.appBar,
  },
};
