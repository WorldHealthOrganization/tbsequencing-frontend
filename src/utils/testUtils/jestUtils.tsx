import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
