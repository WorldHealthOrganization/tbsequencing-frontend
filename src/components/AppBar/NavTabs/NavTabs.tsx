/** @jsxImportSource @emotion/react */
import React, { memo } from 'react';
import { Tabs, Tab } from '@mui/material';
import * as styles from './styles';
import { ITabsConfig } from '../tabsConfig';

interface Props {
  onTabChange: (event: React.SyntheticEvent, value: number) => void;
  activeTab: number;
  tabsConfig: ITabsConfig
  routeMatched: boolean;
}

export const NavTabs = ({
  onTabChange, activeTab, tabsConfig, routeMatched,
}: Props) => {
  const value = routeMatched && activeTab !== -1 ? activeTab : false;

  return (
    <Tabs
      sx={styles.sxTabsStyles}
      value={value}
      onChange={onTabChange}
    >
      {tabsConfig.map((tab, index) => (
        <Tab
          key={tab.label}
          sx={styles.sxTabStyles}
          value={index}
          label={tab.label}
        />
      ))}
    </Tabs>
  );
};
export default memo(NavTabs);
