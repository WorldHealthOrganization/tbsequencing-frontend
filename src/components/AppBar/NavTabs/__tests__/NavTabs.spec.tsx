import React from 'react';
import { getTreeJSONWithRedux, render } from '../../../../utils/testUtils/jestUtils';
import NavTabs from '../NavTabs';

describe('NavTabs component test', () => {
  const mockHandler = jest.fn();
  const mockActiveTab = 0;
  const mockRouteMatched = true;
  const mockTabsConfig = [{
    element: <div />,
    path: '/mock',
    label: 'mock',
  }];

  test('renders NavTabs component without error', () => {
    render(
      <NavTabs
        activeTab={mockActiveTab}
        routeMatched={mockRouteMatched}
        onTabChange={mockHandler}
        tabsConfig={mockTabsConfig}
      />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <NavTabs
        activeTab={mockActiveTab}
        routeMatched={mockRouteMatched}
        onTabChange={mockHandler}
        tabsConfig={mockTabsConfig}
      />,
    );

    expect(tree).toMatchSnapshot();
  });
});
