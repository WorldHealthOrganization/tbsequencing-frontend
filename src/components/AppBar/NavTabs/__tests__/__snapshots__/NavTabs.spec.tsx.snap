// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavTabs component test Should render correctly 1`] = `
<div
  className="MuiTabs-root css-1oujr26-MuiTabs-root"
>
  <div
    className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
    onScroll={[Function]}
    style={
      Object {
        "marginBottom": -0,
        "overflow": "hidden",
      }
    }
  >
    <div
      aria-orientation={null}
      className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-selected={true}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-dpb51x-MuiButtonBase-root-MuiTab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={0}
        type="button"
      >
        mock
        <span
          className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
          style={Object {}}
        />
      </button>
    </div>
  </div>
</div>
`;
